libraries/SCORM_API_wrapper.js

```diff
@@ -24,820 +24,901 @@ further modified by Philip Hutchison
+        var success = false,
+            scorm = pipwerks.SCORM,
+            exitStatus = scorm.data.exitStatus,
+            completionStatus = scorm.data.completionStatus,
+            trace = pipwerks.UTILS.trace,
+            makeBoolean = pipwerks.UTILS.StringToBoolean,
+            debug = scorm.debug,
+            traceMsgPrefix = "SCORM.connection.terminate ";
 
-pipwerks.SCORM.connection.terminate = function(){
 
-    var success = false,
-        scorm = pipwerks.SCORM,
-        exitStatus = scorm.data.exitStatus,
-        completionStatus = scorm.data.completionStatus,
-        trace = pipwerks.UTILS.trace,
-        makeBoolean = pipwerks.UTILS.StringToBoolean,
-        debug = scorm.debug,
-        traceMsgPrefix = "SCORM.connection.terminate ";
+        if (scorm.connection.isActive) {
 
+            var API = scorm.API.getHandle(),
+                errorCode = 0;
 
-    if(scorm.connection.isActive){
+            if (API) {
 
-        var API = scorm.API.getHandle(),
-            errorCode = 0;
+                if (scorm.handleExitMode && !exitStatus) {
 
-        if(API){
+                    if (completionStatus !== "completed" && completionStatus !== "passed") {
 
-             if(scorm.handleExitMode && !exitStatus){
-
-                if(completionStatus !== "completed" && completionStatus !== "passed"){
+                        switch (scorm.version) {
+                            case "1.2":
+                                success = scorm.set("cmi.core.exit", "suspend");
+                                break;
+                            case "2004":
+                                success = scorm.set("cmi.exit", "suspend");
+                                break;
+                        }
 
-                    switch(scorm.version){
-                        case "1.2" : success = scorm.set("cmi.core.exit", "suspend"); break;
-                        case "2004": success = scorm.set("cmi.exit", "suspend"); break;
-                    }
+                    } else {
 
-                } else {
+                        switch (scorm.version) {
+                            case "1.2":
+                                success = scorm.set("cmi.core.exit", "logout");
+                                break;
+                            case "2004":
+                                success = scorm.set("cmi.exit", "normal");
+                                break;
+                        }
 
-                    switch(scorm.version){
-                        case "1.2" : success = scorm.set("cmi.core.exit", "logout"); break;
-                        case "2004": success = scorm.set("cmi.exit", "normal"); break;
                     }
 
                 }
 
-            }
+                //Ensure we persist the data for 1.2 - not required for 2004 where an implicit commit is applied during the Terminate
+                success = (scorm.version === "1.2") ? scorm.save() : true;
```

Comment:
If it helps anyone to review, this is really the main change to this file.

The other big change is that the entire file has been converted into a module, you can see that change in the original repo [here](https://github.com/pipwerks/scorm-api-wrapper/commit/82e455b4032ee08febf64d2fa2bf1aacaebaa446). It was this change which necessitated a minor amend to wrapper.js